packages/smooth_app/lib/pages/product/picture_capturer.dart

```diff
@@ -0,0 +1,165 @@
+          return Scaffold(
+            backgroundColor: Colors.black,
+            appBar: AppBar(),
+            body: Stack(
+              children: <Widget>[
+                Positioned(
+                  child: Align(
+                    alignment: Alignment.center,
+                    child: Image.file(picture!),
+                  ),
+                ),
+                Positioned(
+                  child: Align(
+                    alignment: Alignment.bottomCenter,
+                    child: Padding(
+                      padding: const EdgeInsets.only(bottom: MEDIUM_SPACE),
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceAround,
+                        children: <Widget>[
+                          SmoothActionButton(
+                              text: appLocalizations.retake_photo_button_label,
+                              onPressed: () async {
+                                await _getImage();
+                                setState(() {});
+                              }),
+                          SmoothActionButton(
+                              text: _getConfirmButtonText(
+                                context,
+                                widget.imageType,
+                              ),
+                              onPressed: () async {
+                                _uploadPicture();
+                              }),
+                        ],
+                      ),
+                    ),
+                  ),
+                ),
+              ],
+            ),
+          );
+        });
+  }
+
+  String _getConfirmButtonText(BuildContext context, ImageField imageType) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    switch (imageType) {
+      case ImageField.FRONT:
+        return appLocalizations.confirm_front_packaging_photo_button_label;
+      case ImageField.INGREDIENTS:
+        return appLocalizations.confirm_ingredients_photo_button_label;
+      case ImageField.NUTRITION:
+        return appLocalizations.confirm_nutritional_facts_photo_button_label;
+      case ImageField.PACKAGING:
+        return appLocalizations.confirm_recycling_photo_button_label;
+      case ImageField.OTHER:
+        return appLocalizations.confirm_other_interesting_photo_button_label;
+    }
+  }
+
+  Future<void> _uploadPicture() async {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final SendImage image = SendImage(
+      lang: ProductQuery.getLanguage(),
+      barcode: widget.barcode,
+      imageField: widget.imageType,
+      imageUri: picture!.uri,
+    );
+    final Status result = await OpenFoodAPIClient.addProductImage(
+      ProductQuery.getUser(),
+      image,
+    );
+    if (result.error != null) {
+      // Image upload failed :( Show an error and go back to [AddNewProductPage].
+      await showDialog<void>(
```

Comment:
Please use both `LoadingDialog.run` and `LoadingDialog.error` for that.

Reply:
Thanks, I didn't know this exists.